### DISCUSSION TITLE
How to include calibration into the server

### DISCUSSION THREAD

**Discussion by LeidyMoraV** [PlusToolkit/PlusLib#1170] [Q&A] [answered]
Hello, 

I'm working with Ascension Trackstar, Siemens P300 ultrasound, and the Epiphan frame grabber VGA2USB, and I linked all three hardware through PLUS to Slicer with one config file, then I did another config file in order to do the calibration, now I don't know how to include the calibration matrixes in the server config file so I can visualize everything in 3D Slicer.

**Comment by Sunderlandkyl** [comment]
You can include the CoordinateDefinitions in your config file to include the calibrations.

Ex:
https://github.com/PlusToolkit/PlusLibData/blob/d8f1456bd125bcd7e577805cd57fad1ec9cbfb70/ConfigFiles/Robarts/PlusDeviceSet_Server_NDIAurora.xml#L61-L77

What types of calibrations are you trying to send?

**Reply by LeidyMoraV** [reply]
I'm doing a stylus, phantom and spatial calibration. I've include the results as you mentioned but when I visualize everything in 3D Slider I noticed that the Reference and Probe markers are 'connected' like if the movement was the same in both markers, is this normal? Because when I'm working only with Ascension it seems like the three markers have independent movement 

Here I have the final config file for the server with the calibration matrix included: 
https://pennstateoffice365-my.sharepoint.com/:u:/g/personal/lfm5705_psu_edu/EcdegSS8LOlFhxfEQ5dW4C8B_U7OmPKAeTn9KKKbCM4xcw?e=msePDT

**Reply by Sunderlandkyl** [reply]
What does the transform hierarchy in your Slicer scene look like?

**Reply by LeidyMoraV** [reply]
![image (1)](https://github.com/PlusToolkit/PlusLib/assets/162748695/71f8ed89-3670-402e-b3fd-f220be7e6403)


![image (2)](https://github.com/PlusToolkit/PlusLib/assets/162748695/607acc7c-28e2-419f-92cd-55b108649c63)

**Reply by Sunderlandkyl** [reply, marked as answer]
I think the issue may be that the config file has the ImageToReference transform specified:
```
    <Transform From="Image" To="Reference"
      Matrix="
        0.2 0.0 0.0 0.0
        0.0 0.2 0.0 0.0
        0.0 0.0 0.2 0.0        
        0 0 0 1" />
```

This should be removed as the transform is not static.

**Reply by LeidyMoraV** [reply]
Thank you it worked! Additionally to your correction, I add a: 
`<Image Name="Image" EmbeddedTransformToFrame="Image" /> `
